### DISCUSSION TITLE
Version for Z-endstop on ender3 S1

### DISCUSSION THREAD

**Discussion by thehype1234** [mriscoc/Ender3V2S1#55] [Q&A] [answered]
Hi there. I have massive problems with my mesh level from cr-touch.
On stock and on custom the same.

So I thought about disassembling cr touch and installing the z-endswitch.

Can I use one of the custom firmwares for this, or should I go back to stock for that? 

Thanks

**Comment by mriscoc** [comment]
You cannot use a z-end stop switch with the released versions nor with the stock firmware, you must use the configuration files for manual mesh version, make the changes that refer to the board and connections from the S1 configuration files and compile your own version.

**Reply by thehype1234** [reply]
But there is a stock firmware with z-axis limit switch labeled... 🤔 it's the first firmware on the side 16.12.2021 I think.

I have no clue about compiling... 😅 

I thought I could use the ender3 v2 firmware maybe!? Or is it another board and there are conflicts with it? Maybe you could consider adding a setting to disable the cr touch and using the endstop instead for a future release? It's asked much, I know... I also would contribute some money at a monthly rate, if there's a possibility to...

**Reply by mriscoc** [reply]
I didn't know that there was an official firmware for the S1 without the CRTouch. The firmware for the Ender 3V2 doesn't work for the Ender 3S1. I could generate a special configuration file here: https://github.com/mriscoc/Marlin_Configurations
Thank you, any contribution is welcome.

**Reply by thehype1234** [reply]
Thanks... would be awesome! Many people in Ender e s1 groups seem to have problems with mesh level. Did you experienced it also? Mine does a mesh after I leveled the bed, then I use this mesh in my start gcode... the z motors also move on printing in a layer, but it's off... I think the crTouch homing before the print saves other parameters as it has done on mesh level... hope you understand what I mean!?

**Reply by thehype1234** [reply]
Where I could contribute? Do you have a link?

**Reply by mriscoc** [reply]
At the top of the page there is a link to sponsor the project.

**Reply by thehype1234** [reply]
Found it! 👍

**Reply by mriscoc** [reply]
Thanks, i will to generate the configuration but you must to help me with the test.

**Reply by thehype1234** [reply]
It would be a honor to test it. But I need instructions to get the configuration file in my printer! 😅

**Reply by thehype1234** [reply]
Another quick question... what are the differences between your ender3v2s1 firmware and the ender3s1 firmware? Are there significant settings different?

**Reply by mriscoc** [reply]
The Ender3v2s1 is an advanced firmware with many professional options. The Ender3s1 is a more standard firmware less options for people coming from stock that do not need advanced features.

**Comment by mriscoc** [comment]
I want to know if you saw any guide for z end stop connection for the Ender3S1. I think that the firmware that you saw in Creality site is not for z end-stop, it is for correct a bug about the maximum limit of Z.

**Reply by thehype1234** [reply]
Hi there... sorry for being late. Destroyed my printer trying to connect the endstop. The new printer has arrived today. The connection for the endstop is on the mainboard... it is mentioned in the manual (board layout)

**Reply by thehype1234** [reply]
Hello again...

I checked the version on creality site, it's for the optional z-endstop switch. Without this firmware it doesn't work at all. 

Do you have found time to make a configuration within your firmware for the endstop? 

I've attached the picture from the manual where to put the cable on the motherboard. And the switch itself has to go on the left side of the printer, under the z axis.![image](https://user-images.githubusercontent.com/11811235/156001458-3115c679-9931-47f1-a13f-71c41449bf7e.jpeg)

**Comment by mriscoc** [comment]
Hi, were you able to test that stock firmware in your S1? Is it working fine?
We were trying with another user in the Telegram group, and he was able to use the z end stop but disconnecting the filament sensor, connecting the end stop in that socket and compile the firmware to use those pins. We aren't able to use the z limit interface of the board, but maybe he had a faulty board.

Please test this firmware, it is for Ender3S1 Manual Mesh: DELETED

Check If the Z end stop function is inverted.

**Reply by thehype1234** [reply]
Unfortunately the z endstop is triggered all the time... no change on triggering it manually or leave it alone. Endstop is connected to board btw...

**Reply by thehype1234** [reply]
Oh and the stock firmware is working good so far... but no possibility to do a manual mesh. 🤷‍♂️😉 would be awesome if you can do something for this endstop.

**Reply by thehype1234** [reply]
![image](https://user-images.githubusercontent.com/11811235/156070361-c66ff12a-2711-40d7-b3f3-9332ca133e6c.jpeg)

**Comment by mriscoc** [comment]
Thank you for your test. If you disconnect the z end stop, the End-stops diagnostic shows also TRIGGERED?

**Reply by thehype1234** [reply]
Yes it does... do you have documentation for marlin, which pins used on board for this? Somebody told me this is the tricky part to get any documentation for the pins on this new board...!? Thanks for your work!

**Reply by thehype1234** [reply]
Creality has released the 3s1 firmware on GitHub...

**Reply by mriscoc** [reply]
Hooray! We are porting the configuration values; it is almost finished. As I suspected, they are using a different PIN for the Z end-stop. Be ready to do some test!

**Reply by mriscoc** [reply]
Please, test this version:
[Ender3S1-ManualMesh-20220301.zip](https://github.com/mriscoc/Ender3V2S1/files/8165618/Ender3S1-ManualMesh-20220301.zip)

**Reply by thehype1234** [reply]
Z-endstop is working in this release... unfortunately on manual mesh level it seems to not work 100% correct. This has nothing to do with the switch itself... if you set the offset for a certain point with paper method, and let's say z is at 0.01 the paper is free. You go to 0.00 paper is still free. Then you go to -0.01 the nozzle grabs the paper. So far so good, but if you dial it back to 0.00 the paper is not free again until you dial it another step back to 0.01.

Hope it is understandable!? Seems to me that this setting only changes on every second step you make... is this even possible? 

But so far, great work! 😉

**Reply by mriscoc** [reply]
Maybe that is a backlash problem, the Z doesn't go to the same point if the movement is from above or from below the point. 0.01 mm is more that the mechanical precisión of the printer and is better if the first layer is near to 0.3mm.

**Reply by DadoKomar** [reply]
> Please, test this version: [Ender3S1-ManualMesh-20220301.zip](https://github.com/mriscoc/Ender3V2S1/files/8165618/Ender3S1-ManualMesh-20220301.zip)

Hello sir. 

First of all, great job with firmware. I do like tunning possibility during printing.

I am also into Z-switch testing. 
Q1. Would you be so kind and compile for me some last version for z switch? I am not into compiling, sorry.

I will plug z switch into mainboard. (Anaway I am also not using filament sensor).
Q2. Do I understand it right, then I can remove that bloody sensor from head?

Q3. Is possible to add one setting into menu to switch between sensor and Z end switch? Now when you have that extra input known, should not be hard, no?

I am gonna become your patreon. Thanks again, best firmware!

**Reply by mriscoc** [reply]
Hi. It is not possible to have Manual Mesh and Probe leveling enabled at the same time.
The Ender3S1 comes with a spare end-stop to be connected in the Z end-stop connector, the S1 manual mesh firmware uses that configuration.
You can sponsor that configuration here: https://github.com/mriscoc/Special_Configurations

**Reply by DadoKomar** [reply]
Will try. Just will take some time to get first compile...

**Comment by mriscoc** [comment, marked as answer]
I did post the configuration files for the S1 manual mesh version in the repository: https://github.com/mriscoc/Marlin_Configurations
Now you can compile your firmware.

**Comment by jeffibacache** [comment]
Hey, I had the exact same issue. Replaced my CR-Touch with a mechanical limit switch and during Auto Home Z would just raise and sit there doing nothing.

Turns out the problem is Z_MIN_ENDSTOP_HIT_STATE being set to HIGH in Configuration.h. A standard NO switch keeps the pin HIGH when not pressed, so Marlin thinks the endstop is always triggered and refuses to move Z down.

Fix is simple, just change it to LOW:

#define Z_MIN_ENDSTOP_HIT_STATE LOW

Reflashed and Auto Home is working perfectly now. Z goes all the way down and stops right on the switch like magic ❤️❤️
